presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1885,89 +1885,213 @@ private void computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope sou
-            if (fields.isEmpty()) {
-                if (starPrefix.isPresent()) {
-                    throw semanticException(TABLE_NOT_FOUND, allColumns, "Table '%s' not found", starPrefix.get());
+                List<Field> fields = (List<Field>) scope.getRelationType().getVisibleFields();
+                if (fields.isEmpty()) {
+                    if (!node.getFrom().isPresent()) {
+                        throw semanticException(COLUMN_NOT_FOUND, allColumns, "SELECT * not allowed in queries without FROM clause");
+                    }
+                    throw semanticException(COLUMN_NOT_FOUND, allColumns, "SELECT * not allowed from relation that has no columns");
                 }
-                if (!node.getFrom().isPresent()) {
-                    throw semanticException(COLUMN_NOT_FOUND, allColumns, "SELECT * not allowed in queries without FROM clause");
+
+                analyzeAllColumnsFromTable(fields, allColumns, node, scope, outputExpressionBuilder, selectExpressionBuilder, scope.getRelationType());
+            }
+        }
+
+        private void analyzeAllColumnsFromTable(
+                List<Field> fields,
+                AllColumns allColumns,
+                QuerySpecification node,
+                Scope scope,
+                ImmutableList.Builder<Expression> outputExpressionBuilder,
+                ImmutableList.Builder<SelectExpression> selectExpressionBuilder,
+                RelationType relationType)
+        {
+            if (!allColumns.getAliases().isEmpty()) {
+                if (allColumns.getAliases().size() != fields.size()) {
+                    throw semanticException(
+                            MISMATCHED_COLUMN_ALIASES,
+                            allColumns.getAliases().get(0),
+                            "Column alias list has %s entries but subquery has %s columns",
+                            allColumns.getAliases().size(),
+                            fields.size());
                 }
-                throw semanticException(COLUMN_NOT_FOUND, allColumns, "SELECT * not allowed from relation that has no columns");
             }
 
-            for (Field field : fields) {
+            ImmutableList.Builder<Field> itemOutputFieldBuilder = ImmutableList.builder();
+
+            for (int i = 0; i < fields.size(); i++) {
+                Field field = fields.get(i);
                 int fieldIndex = relationType.indexOf(field);
-                FieldReference expression = new FieldReference(fieldIndex);
-                outputExpressionBuilder.add(expression);
-                ExpressionAnalysis expressionAnalysis = analyzeExpression(expression, scope);
+                FieldReference fieldReference = new FieldReference(fieldIndex);
+                outputExpressionBuilder.add(fieldReference);
+                selectExpressionBuilder.add(new SelectExpression(fieldReference, Optional.empty()));
+                if (!allColumns.getAliases().isEmpty()) {
```

Comment:
I'd structure this as:

```java
Optional<String> alias;
if (!allColumns.getAliases().isEmpty()) {
    alias = Optional.of((allColumns.getAliases().get(i)).getValue());
}
else {
    alias = field.getName();
}

itemOutputFieldBuilder.add(new Field(
        field.getRelationAlias(),
        alias,
        field.getType(),
        false,
        field.getOriginTable(),
        field.getOriginColumnName(),
        alias.isPresent()));
```

Reply:
Are you sure that `alias.isPresent()` is the right value for `aliased` parameter in the case  there were no column aliases?

Reply:
It is wrong actually. `TestDescribeOutput` fails.